---
title: "Exploratory data analysis of genomic data from breast cancer studies"
author: "Urminder Singh"
date: "December 8, 2018"
output: pdf_document
---

```{r setup, include=FALSE, cache=FALSE,message=FALSE}
#knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(echo = TRUE,message = FALSE,warning = FALSE,cache = F)

#if not installed --> install

#required packages
library(TCGAbiolinks)
library(dplyr)
library(DT)
library(data.table)
library(plyr)
library(maftools) 
library(readr)
library(ggplot2)
library(gganatogram)
library(viridis)
library(gridExtra)
library(grid)
library(RColorBrewer)

```



# Introduction

Our data consists of gene expression data from various samples collected to study breast cancer (Breast Invasive Carcinoma). These samples were collected from a number of different individuals. Then, the samples' RNA were extracted sequenced to get RNA-seq reads from the sample. Using the RNA-seq, the gene expression was estimated. However, we used the data from (Wang et. al. 2018) where the expression data was mapped and normalized again to remove batch effects, which made the dataset comparable across the samples.
Apart from the gene expression data for each sample, we also have associated metadata for each sample. This metadata provides information about the individual, tissue, disease etc.associated with each sample.

Our dataset has expression estimates of around 20,000 genes across 1,092 samples. Out of these 1,092 samples, 982 had tumor and 110 were normal tissues without tumor. We also added the gene metadata to our dataset which describes the gene features and functions. This also includes mutation information of the genes. This information can provide crucial information about how different mutations in the genome can lead to cancer.

Overall, our dataset is GxS matrix where each row corresponds to a gene and each column corresponds to a RNA-seq sample. Additionally, we have metadata for all the rows and all the columns.


# Data

We used data from TCGA. TCGA is a comprehensive repository of human cancer molecular and clinical data, TCGA database has collected clinical and molecular phenotypes of thousands of tumor patients across different tumor types. The TCGA dataset, contains:

* Clinical information about participants
* Metadata about the samples (e.g. the weight of a sample portion, etc.)
* Histopathology slide images from sample portions
* Molecular information derived from the samples (e.g. mRNA/miRNA expression, protein expression, copy number, etc.)



First, we collected data clinical data from TCGA for breast cancer (BRCA) studies. This data contains
Then we downloaded the mutation information for the BRCA studies. This data
Finally, I collected the gene expression data  from (Wang et. al. 2018). This data contains the estimated expression values of genes in the BRCA study. This data was normalized and batch corrected so it is comparable across different TCGA samples.


## Data downloading and processing

This section describes how the data was downloaded and pre-processed.

### Downloading clinical data from TCGA

We used the TCGABiolinks package to download TCGA data. The function *GDCquery_clinic* is used to download the data. We wrote additional functions to clean the data and arrange it into a dataframe.

```{r,message=FALSE}
colsToKeep<-c("clinical.submitter_id","clinical.classification_of_tumor",
              "clinical.primary_diagnosis","clinical.tumor_stage",
              "clinical.age_at_diagnosis","clinical.vital_status",
              "clinical.days_to_death","clinical.tissue_or_organ_of_origin",
              "clinical.days_to_birth","clinical.site_of_resection_or_biopsy",
              "clinical.days_to_last_follow_up","clinical.cigarettes_per_day",
              "clinical.weight","clinical.alcohol_history","clinical.bmi",
              "clinical.years_smoked","clinical.height","clinical.gender",
              "clinical.year_of_birth","clinical.race","clinical.ethnicity",
              "clinical.year_of_death","clinical.bcr_patient_barcode",
              "clinical.disease","submitter_id","sample_type","tissue_type",
              "portions.submitter_id","portions.analytes.analyte_type",
              "portions.analytes.submitter_id","portions.analytes.analyte_type_id",
              "portions.analytes.aliquots.analyte_type","portions.analytes.aliquots.submitter_id")
#Function takes a df and expands it by unlisting elements at a column
expand<-function(df,colName){
  res<-data.frame()
  #for each row
  for(i in 1: dim(df)[1]){
    thisRow<-df[i, ! (colnames(df) %in% c(colName))]
    tempdf<-as.data.frame(df[i, c(colName)])
    #if list is empty skip that row
    if(dim(tempdf)[1]<1){
      next
    }
    #change colnames so they are unique
    colnames(tempdf)<-paste(paste(colName,".",sep = ""),colnames(tempdf),sep = "")
    #print(paste(i,colnames(tempdf)))
    newRow<-cbind(thisRow,tempdf,row.names = NULL)
    res<-bind_rows(res,newRow)
    
  }
  #print(res)
  return(res)
}

getjoinedBiospcClinc<-function(projName){
  print(paste("Downloading",projName))
  clinicalBRCA <- GDCquery_clinic(project = projName, type = "clinical")
  biospecimenBRCA <- GDCquery_clinic(project = projName, type = "Biospecimen")
  
  #rename all cols from clinical table with suffix clinical
  colnames(clinicalBRCA)<- paste0("clinical.",colnames(clinicalBRCA))
  
  #expand biospecimen data in the order portions, portions.analytes, portions.analytes.aliquots
  toUnpack<-c("portions", "portions.analytes", "portions.analytes.aliquots")
  for(s in toUnpack){
    biospecimenBRCA<-expand(biospecimenBRCA,s)
  }
  #add patient barcode to biospecimen data
  biospecimenBRCA<- biospecimenBRCA %>% mutate(clinical.bcr_patient_barcode=substr(submitter_id,1,nchar(as.character(submitter_id))-4))
  #join clinical and biospecimen
  finalJoined<-join(clinicalBRCA,biospecimenBRCA,by="clinical.bcr_patient_barcode")
  return(finalJoined)
}
###################################################################
projName<-"BRCA"
##Download only BRCA metadata
brcaDF<-getjoinedBiospcClinc(paste("TCGA",projName,sep = "-"))
brcaDF<-brcaDF[,colsToKeep]
#remove cols with all NA values
naCols<-colnames(brcaDF)[sapply(brcaDF, function(x)all(is.na(x)))]
brcaDF<-brcaDF[,!(colnames(brcaDF) %in% naCols)]

head(brcaDF[,1:5],3)
```

The clinical data for BRCA studies is stored in a dataframe *brcaDF*. 

### Downloading mutation data from TCGA

Next, we downloaded mutation data using TCGABiolinks. After getting the mutation data, we joined the clinical table with the mutation table to have clinical information for each mutation in the mutation table.

```{r,message=FALSE,warning=FALSE,results="hide"}

brcaMAF <- GDCquery_Maf(projName, pipelines = "varscan2")
#join mutation information with the clinical data
colnames(brcaDF)[which(colnames(brcaDF)==
                         "portions.analytes.aliquots.submitter_id")]="Tumor_Sample_Barcode"
brcaMAF_MD<-join(brcaMAF,brcaDF)
```

We now have a data frame of dimensions `r dim(brcaMAF_MD)`. This data frame contains clinical information and coressponding mutation information for each BRCA patient. We will use this data to explore various...

### Downloading gene expression data

Next, we downloaded the gene expression data for the BRCA studies. 

```{r,message=FALSE,warning=FALSE,results="hide"}
#download exp data from github for BRCA samples. reading files can take several minutes
brcaexp_nontumor <- read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/brca-rsem-fpkm-tcga.txt.gz",
                            "\t", escape_double = FALSE, trim_ws =TRUE)
brcaexp_tumor <- read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/brca-rsem-fpkm-tcga-t.txt.gz", 
                         "\t", escape_double = FALSE, trim_ws = TRUE)

#formatting. add row names remove extra columns
rn<-brcaexp_nontumor$Hugo_Symbol
brcaexp_nontumor<-brcaexp_nontumor[,3:dim(brcaexp_nontumor)[2]]
rownames(brcaexp_nontumor)<-rn

rn<-brcaexp_tumor$Hugo_Symbol
brcaexp_tumor<-brcaexp_tumor[,3:dim(brcaexp_tumor)[2]]
rownames(brcaexp_tumor)<-rn

```

We now have two expression datasets for tumor and normal or non-tumor tissues from BRCA studies. These datasets contains the gene expression patterns over the tumor samples and normal samples. Each row corresponds to a gene and each column corresponds to a sample. The data dimentions for tumor expression data is `r dim(brcaexp_tumor)` and for normal expression data is `r dim(brcaexp_nontumor)`


Finally, our data consist of three data frames

  1. *brcaMAF_MD* contains all the mutation information
  2. *brcaexp_nontumor* contains the gene expression values over normal samples from TCGA
  3. *brcaexp_tumor* contains the gene expression values over tumor samples from TCGA


# Analysis

First, we looked at the mutation data to find which genes are highly mutated in BRCA and what types of different mutations are present in the data. We used the package *maftools* to do this analysis.

```{r,message=FALSE,warning=FALSE,results="hide"}
#function to take a list of maf files and summarize them
plotmafSummaryList <- function(mafList) {
l1 <- mafList
#for each item in list do calculations
lnames <- names(l1)
plotList <- list()
k <- 1
plist <- list()
for (i in lnames) {
print((i))
print(dim(l1[[i]]))
if (dim(l1[[i]])[1] < 1) {
next
}
#plot summary and save to pdf
maf <- read.maf(l1[[i]], isTCGA = T)
plotmafSummary(
maf = maf,
rmOutlier = TRUE,
addStat = 'median',
dashboard = T,
titvRaw = FALSE,
showBarcodes = F,
top = 10
)
mtext(
paste("Mutation summary by", i),
outer = T,
cex = NA,
line = -1.0,
side = 3
)
}

}

#split maf file into categories by a given variable in the clinical metadata file
splitMafby <- function(clinicalData, by, mafData) {
#get tumor samps fo different values of by
myList <- list()
uniqVals <- clinicalData %>% select(by) %>% unique
for (i in 1:nrow(uniqVals)) {
#i<-1
s <- as.character(uniqVals[i, 1])
print(s)
tList <-
clinicalData %>% filter(clinicalData[, by] == s) %>% select(Tumor_Sample_Barcode) %>%
unique
thisData <-
mafData %>% filter(Tumor_Sample_Barcode %in% tList$Tumor_Sample_Barcode)
myList[[s]] <- thisData
}

return(myList)
}

#function to plot top mutated genes with ggplot
plotGeneVarFreq <- function(mafList) {
#for each item in list
lnames <- names(mafList)
plotList <- list()
k <- 1
plist <- list()
getPalette = colorRampPalette(brewer.pal(9, "Paired"))
# assign colors manually to be consistent with missing data
variants <- c(
"Missense_Mutation",
"Silent",
"3'UTR",
"Nonsense_Mutation",
"5'Flank",
"Intron",
"Splice_Region",
"RNA",
"5'UTR",
"Splice_Site",
"In_Frame_Del",
"Frame_Shift_Ins",
"Frame_Shift_Del",
"In_Frame_Ins",
"3'Flank",
"Nonstop_Mutation",
"Translation_Start_Site",
"IGR"
)
colourCount = length(unique(variants))
palette <- getPalette(colourCount)
names(palette) <- variants

for (i in lnames) {
print((i))
print(dim(l1[[i]]))
if (dim(l1[[i]])[1] < 1) {
next
}

#find top mutated genes
thisMaf <- mafList[[i]]
topGenes <-
thisMaf %>% filter(Variant_Classification != "Silent") %>% select(Hugo_Symbol) %>% group_by(Hugo_Symbol) %>% count %>% arrange(desc(freq)) %>% top_n(n=10)
maftest <- thisMaf %>% filter(Hugo_Symbol %in% topGenes$Hugo_Symbol & Variant_Classification != "Silent") %>% select(Hugo_Symbol, Variant_Classification)


p <-
ggplot(data = maftest, aes(x = Hugo_Symbol, fill = Variant_Classification)) +
geom_bar(stat = "count") + scale_x_discrete(limits = rev(topGenes$Hugo_Symbol)) +
coord_flip() +
theme(legend.position = "right") +
theme(
axis.text.x = element_text(size = 11, face = "bold"),
axis.text.y = element_text(size = 11, face = "bold"),
panel.grid.major = element_blank(),
panel.grid.minor = element_blank(),
panel.border = element_blank(),
panel.background = element_blank(),
axis.title = element_text(size = 12, face = "bold")
) +
ylab("") + xlab("") + scale_fill_manual(values = palette) + ggtitle(i)

plist[[k]] <- p
k = k + 1

}
#arrange plots on a grid
do.call("grid_arrange_shared_legend", c(plist))


}

#plot on a grid with single legend
#function reference https://stackoverflow.com/questions/13649473/add-a-common-legend-for-combined-ggplots
grid_arrange_shared_legend <- function(...) {
plots <- list(...)
g <-
ggplotGrob(plots[[1]] + theme(legend.position = "bottom"))$grobs
legend <- g[[which(sapply(g, function(x)
x$name) == "guide-box")]]
lheight <- sum(legend$height)
grid.arrange(
do.call(arrangeGrob, lapply(plots, function(x)
x + theme(legend.position = "none"))),
legend,
ncol = 1,
heights = unit.c(unit(1, "npc") - lheight, lheight)
)
}

```

```{r fig1, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig1}plotting example"}
#plot summary for full dataset
tempList<-list()
tempList[["BRCA"]]<-brcaMAF
plotmafSummaryList(tempList)

```

We found that

Next we looked at how the gene mutations differ for different categories e.g gender, race cancer type etc. First, we looked at which genes were mutated for samples from different race.

```{r fig2, fig.width=10, fig.height=12,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig2}plotting examplerace"}
#split the maf data by clinical.race and plot summary for each category
l1<-splitMafby(brcaDF,"clinical.race",brcaMAF)
#plot only white black and asian as others are not reported
l1<-l1[c("white","black or african american","asian" )]
plotGeneVarFreq(l1)

```

We found that

Then, we looked at how genes are mutated for top two cancer types i.e. ductal carcinoma and lobular carcinoma.

```{r fig3, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide"}
#split the maf data by clinical.race and plot summary for each category
filterList<- brcaDF %>% select(clinical.primary_diagnosis) %>% count %>% arrange(desc(freq)) %>% top_n(n=2)
l2<-splitMafby(brcaDF%>%filter(clinical.primary_diagnosis %in% filterList$clinical.primary_diagnosis),"clinical.primary_diagnosis",brcaMAF)
plotGeneVarFreq(l2)
```

Finally, we looked at if there is a difference in the set of genes mutated by gender.

```{r fig4, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide"}
#split the maf data by clinical.race and plot summary for each category
l3<-splitMafby(brcaDF,"clinical.gender",brcaMAF)
plotGeneVarFreq(l3)
```

From the analysis above we noticed that BRCA1 and BRCA2 are not in the set of top mutated genes. These genes are known to be associated with cancer and a mutation in these genes can increase the risk of breast and ovarian cancer in women. We then compared how many mutations were associated with BRCA1 and BRCA2.

```{r fig5, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig2}plotting exampleBRCA"}
brcaMAF_MD_geneGroupBRCA<-brcaMAF_MD%>%select(Hugo_Symbol)%>%mutate(geneName=ifelse(Hugo_Symbol=="BRCA1","BRCA1",ifelse(Hugo_Symbol=="BRCA2","BRCA2","Others"))) %>%select(geneName)%>% group_by(geneName) %>% count %>% mutate(logFreq=log(freq))

p1<-ggplot(data=brcaMAF_MD_geneGroupBRCA, aes(x=geneName,y=logFreq,fill=geneName))+geom_bar(stat = "identity")+theme(legend.position = "none")+
  theme(axis.text.x = element_text(size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_brewer(palette="Paired")



p2<-ggplot(data=brcaMAF_MD%>%filter(Hugo_Symbol %in% c("BRCA1","BRCA2")),aes(x=Hugo_Symbol,fill=Variant_Classification))+geom_bar()+
  theme(axis.text.x = element_text(size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),legend.text=element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_brewer(palette="Dark2")

grid.arrange(p1, p2, nrow = 1)

```

From figure we see that ...


After looking at various mutated genes, we looked at the expression pattern of these genes for tumor and normal samples.

```{r fig6, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig6}plotting express"}
#plot box plots of top mutated genes
topGenes<-c("PIK3CA","TP53","TTN","GATA3","CDH1","MAP3K1","MUC16","KMT2C","MUC4","PTEN")
nt<-as.data.frame(t(brcaexp_nontumor[topGenes,]))
colnames(nt)<-topGenes
#tumordata
tdata<-as.data.frame(t(brcaexp_tumor[topGenes,]))
colnames(tdata)<-topGenes
nt<-nt%>%mutate(source="Non-Tumor")
tdata<-tdata%>%mutate(source="Tumor")

topCombined<-bind_rows(nt,tdata)
ggplot(melt(topCombined), aes(x=factor(variable),y=value,fill=factor(source)))+geom_boxplot(outlier.colour=NA)+scale_y_log10()+theme(legend.position = "top",legend.text = element_text(size = 15,face = "bold"))+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ 
  scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")


```

From fig we found that the expression of

After that we compiled a list of genes are known to be differentially expressed in cancer.

```{r fig7, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig7}plotting express"}
#ref for data https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5470989/
upGenes<-c("E2F1", "EZH2", "FOXM1", "MYBL2", "PLK1")
dwnGenes<-c("SCARA5", "MYOM1", "NKAPL", "PEG3", "USP2")


ntUP<-as.data.frame(t(brcaexp_nontumor[upGenes,]))
colnames(ntUP)<-upGenes
ntUP<-ntUP%>%mutate(source="Non-Tumor")
tdataUP<-as.data.frame(t(brcaexp_tumor[upGenes,]))
colnames(tdataUP)<-upGenes
tdataUP<-tdataUP%>%mutate(source="Tumor")

diffExpgenes<-bind_rows(ntUP,tdataUP)
dodge <- position_dodge(width = 0.5)

p1<-ggplot(melt(diffExpgenes), aes(x=factor(variable),y=value,fill=factor(source)))+geom_violin(position = dodge)+geom_boxplot(width=.1, outlier.colour=NA, position = dodge)+scale_y_log10()+theme(legend.position = "top")+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")

#for down regulated in cancer
ntDWN<-as.data.frame(t(brcaexp_nontumor[dwnGenes,]))
colnames(ntDWN)<-dwnGenes
ntDWN<-ntDWN%>%mutate(source="Non-Tumor")
tdataDWN<-as.data.frame(t(brcaexp_tumor[dwnGenes,]))
colnames(tdataDWN)<-dwnGenes
tdataDWN<-tdataDWN%>%mutate(source="Tumor")
diffExpgenesDWN<-bind_rows(ntDWN,tdataDWN)


p2<-ggplot(melt(diffExpgenesDWN), aes(x=factor(variable),y=value,fill=factor(source)))+geom_violin(position = dodge)+geom_boxplot(width=.1, outlier.colour=NA, position = dodge)+scale_y_log10()+theme(legend.position = "top")+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")

grid.arrange(p1, p2, nrow = 1)

```

## Comparing expression of mutated BRCA genes with other cancer types

Finally, we wanted to see expression of genes in different cancer types.

```{r,message=FALSE,warning=FALSE,results="hide"}
#download more expression data
#read tumor data for brca,liver,stomach,colon,Lung
brca <- brcaexp_tumor

coad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/coad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
coadRownames<-coad$Hugo_Symbol
coad<-coad[,3:dim(coad)[2]]
rownames(coad)<-coadRownames

lihc<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/lihc-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
lihcRownames<-lihc$Hugo_Symbol
lihc<-lihc[,3:dim(lihc)[2]]
rownames(lihc)<-lihcRownames

luad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/luad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
luadRownames<-luad$Hugo_Symbol
luad<-luad[,3:dim(luad)[2]]
rownames(luad)<-luadRownames

stad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/stad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
stadRownames<-stad$Hugo_Symbol
stad<-stad[,3:dim(stad)[2]]
rownames(stad)<-stadRownames

topGenes<-c("PIK3CA","TP53","GATA3","CDH1","MAP3K1","PTEN")
expVals<-c()
temp<-t(brca[topGenes,])
colnames(temp)<-topGenes
meanbrca<-colMeans(temp)

temp<-t(coad[topGenes,])
colnames(temp)<-topGenes
meancoad<-colMeans(temp)

temp<-t(lihc[topGenes,])
colnames(temp)<-topGenes
meanlihc<-colMeans(temp)

temp<-t(luad[topGenes,])
colnames(temp)<-topGenes
meanluad<-colMeans(temp)

temp<-t(stad[topGenes,])
colnames(temp)<-topGenes
meanstad<-colMeans(temp)
```

```{r fig8, fig.width=10, fig.height=8,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig8}plotting express tissue"}

organs<-c("breast","colon","liver","lung","stomach")
type<-c( "other","digestion","digestion","respiratory","digestion")
colour<-c("#41ab5d","orange","orange","steelblue","orange")


plots<-list() 
for(i in 1:6){
  vals<-c(meanbrca[i],meancoad[i],meanlihc[i],meanluad[i],meanstad[i])
  gganatogramData<-data.frame(organ=organs,type=type,colour=colour,value=as.numeric(vals),stringsAsFactors=F)
  
  p<-gganatogram(data=gganatogramData, fillOutline='#a6bddb', organism='human', sex='female', fill="value")+ 
    theme_void()+  scale_fill_gradient(low = "yellow", high = "red",name= paste(topGenes[i],"(fpkm)"))+theme(legend.text = element_text(size=15,face = "bold"),legend.position = c(0.75, 0.2))
  
  plots[[i]]<-p
}

n <- length(plots)
nCol <- floor(sqrt(n))
do.call("grid.arrange", c(plots, ncol=3))

```

The plots in fig shows that 

# Conclusion

# System information

```{r,tidy=TRUE}
sessionInfo()
```
