---
title: "Exploratory data analysis of genomic data from breast cancer studies"
author: "Carlos Back and Urminder Singh"
date: "December 13, 2018"
output: pdf_document
bibliography: references.bib
---

```{r setup, include=FALSE, cache=FALSE,message=FALSE}
#knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(echo = F,message = FALSE,warning = FALSE,cache = F,tidy.opts=list(width.cutoff=60),tidy=TRUE,fig.pos= "h")
#if not installed --> install
#required packages
library(readr)
library(dplyr)
library(ggplot2)
library(viridis)
library(maps)

```



# Introduction

Cancer is a serious and potentially life-threatning disease. Each year the number of people diagnosed with cancer are increasing and this number is expected to increase rapidly in coming years. In 2018, an estimated 1,735,350 new cases of cancer will be diagnosed  and 609,640 people will die from the disease in the United States alone [@nationalcancerinstitute]. By the year 2030 we expect to see 23.6 million cancer cases per year worldwide. As of 2018, breast cancer is the most common cancer after lung and prostate cancer[@nationalcancerinstitute]. In order to provide effective treatment for cancer, researchers must fully understand  this disease at the genomic level. To do so, The Cancer Genome Atlas (TCGA) project started in 2005 which has a catalogue of genomic data collected from cancer patients [@tcga; @wang2018unifying]. To further understand the prevalence of cancer among different populations, resources like United States Cancer Statistics (USCS), Centers for Disease Control and Prevention (CDC), National Program of Cancer Registries (NPCR), National Center for Health Statistics(NCHS), and National Cancer Institute's (NCI) keep a comprehensive record of cancer cases and associated attributes.

In this project, we analysed data from resources USCS, NPCR, NCHS, NCI and TCGA. In part one, we summarized cancer statistics from the CDC data. This dataset had more than a million cancer cases and cancer deaths for the whole U.S. population from 1999 to 2015, along with information about patients state of residence, states' population, type of cancer, gender and race. We looked at the different trends of various cancer cases to see if they correlate with any other factor like gender, age, geographic location etc. In particular we focused on the incidences of breast cancer accross the United States.

Next, in part two, we collected breast cancer (BRCA) data from TCGA in order to find associations between various clinical attributes, gene mutations, and gene expression. We found that the sets of highly mutated genes in BRCA could be different under different clinical attributes. We observed that the most common type of mutation in all these sets is missense mutation. We also looked at the expression patterns of BRCA associated genes in tumor and normal samples. It appears that some of the higly mutated genes change overall level of expression. Finally, we visually compared mean expression levels of these genes under five different cancers (breast, colon, liver, lung and stomach)


\section*{\centering{Part I}}

```{r, echo=FALSE,results="hide"}
cdc_data <-
  read_delim(
    "https://raw.githubusercontent.com/urmi-21/Stat579Project/master/BYAREA.txt.gz",
    "|",
    escape_double = FALSE,
    trim_ws = TRUE,
    col_types = cols(.default = col_factor(), POPULATION =
                       col_integer())
  )

cdc_data$COUNT <- as.numeric(levels(cdc_data$COUNT))[cdc_data$COUNT]
cdc_data <- na.omit(cdc_data)
by_area <- cdc_data %>%
  filter(
    YEAR != "2011-2015",
    SEX != "Male and Female",
    SITE != "All Cancer Sites Combined",
    RACE != "All Races"
  ) %>%
  select(AREA, COUNT, EVENT_TYPE, POPULATION, RACE, SEX, SITE, YEAR) %>%
  subset()
by_area$YEAR <- droplevels(by_area$YEAR)
by_area$SEX <- droplevels(by_area$SEX)
by_area$RACE <- droplevels(by_area$RACE)
states <- map_data("state")
## Top 5 types of cancer in the U.S. (by incidence)
top5 <- by_area %>%
  filter(EVENT_TYPE == "Incidence") %>%
  group_by(SITE) %>%
  summarize(sum_counts = sum(COUNT)) %>%
  arrange(desc(sum_counts)) %>%
  slice(1:5) %>%
  data.frame()
by_area %>%
  filter(SITE %in% top5$SITE & EVENT_TYPE == "Incidence") %>%
  group_by(SEX, SITE) %>%
  summarize(soma = sum(COUNT)) %>%
  ggplot(aes(SITE, soma, fill = SEX), position = position_stack(reverse = TRUE)) +
  geom_col() +
  coord_flip() +
  theme(legend.position = "bottom") +
  labs(x = "Type of cancer") +
  labs(y = "Cases")
##Number of cases by year
by_area %>%
  filter(SITE %in% top5$SITE & EVENT_TYPE == "Incidence") %>%
  ggplot(aes(
    x = YEAR,
    y = COUNT,
    group = SITE,
    color = SITE
  )) +
  stat_summary(fun.y = sum, geom = "line")
##Incidence by gender
by_area %>%
  filter(SITE %in% top5$SITE & EVENT_TYPE == "Incidence") %>%
  ggplot(aes(
    x = YEAR,
    y = COUNT,
    group = SITE,
    color = SITE
  )) +
  stat_summary(fun.y = sum, geom = "line") +
  facet_grid(SEX ~ .) +
  labs(x = "Type of cancer") +
  labs(y = "Cases")
#BREAST CANCER
byarea_BC <- by_area %>%
  subset(SITE == "Female Breast")
##Incidence of breast cancer by race
byarea_BC %>%
  ggplot(aes(
    x = YEAR,
    y = COUNT / POPULATION * 100000,
    group = RACE,
    color = RACE
  )) +
  stat_summary(fun.y = mean, geom = "line") +
  facet_grid(EVENT_TYPE ~ .) +
  labs(x = "Year") +
  labs(y = "Rate of incidence by 100,000")
```

The results showed that breast cancer is the main cancer type in women and the main cancer type overall. More than that, the number of cases of breast cancer have been increasing along the years, contrary the any other top 5 type of cancer in the country (urinary bladder, colon and rectum, prostate, lung and bronchus). Prostate cancer cases, the main type of cancer in men, is only the third type of cancer in general, behind lung and bronchus. Incidences of breast cancer are higher in white women, followed by black and asian women. In this data, hispanic women have the lower incidences of cancer. Regarding mortality, white and black women have similar rates, about 30 per 100,000 people. All the other races have similar mortality of around 10 per 100,000 people. 

```{r, echo=FALSE,results="hide"}
##Mapping breast cancer in the U.S.
bc_map<-byarea_BC %>% 
  group_by(AREA, EVENT_TYPE) %>% 
  summarize(
    total=mean(COUNT/POPULATION*100000)
  )
bc_map$region <- tolower(bc_map$AREA)
bc_map <- bc_map %>% left_join(states, by="region")

bc_map %>%
  filter(EVENT_TYPE=="Incidence") %>% 
  ggplot(aes(x = long, y = lat, fill = total, group = total)) +
  geom_polygon(colour="white", size = 0.1) +
  labs(x = NULL, 
       y = NULL, 
       title = "Breast Cancer in the U.S.", 
       subtitle = "Incidence Rate by Population") +
  scale_fill_viridis(option = "magma", direction = -1, name = "Rate")

bc_map %>%
  filter(EVENT_TYPE=="Mortality") %>% 
  ggplot(aes(x = long, y = lat, fill = total, group = total)) +
  geom_polygon(colour="white", size = 0.1) +
  labs(x = NULL, 
       y = NULL, 
       title = "Breast Cancer in the U.S.", 
       subtitle = "Mortality Rate by Population") +
  scale_fill_viridis(option = "magma", direction = -1, name = "Rate")
```
The distribution of cases among US states show that the incidence of breast cancer is higher in northern states such as Maine, Vermont, New Hampshire, Montana and North Dakota. Mississipi, Kentucky and West Virginia have a slightly fewer cases. The mortality by breast cancer is also high in northern states, but is even higher in Mississipi, West Virginia and Missouri. 

All these findings shows that we should investigate breast cancer further to find better ways to diagnose and combat the disease. 



\section*{\centering{Part II}}

# Data and Methods

We used data from TCGA [@tcga]. TCGA is a comprehensive repository of human cancer molecular and clinical data, TCGA database has collected clinical and molecular phenotypes of thousands of tumor patients across different tumor types. The TCGA dataset, contains:

* Clinical information about participants
* Metadata about the samples (e.g. the weight of a sample portion, etc.)
* Histopathology slide images from sample portions
* Molecular information derived from the samples (e.g. mRNA/miRNA expression, protein expression, copy number, etc.)   

We downloaded three types of data i.e. clinical data, mutation data and gene expression data. The clinical data contains information about each patient who participated in the study. It includes variables like gender, age, race, cancer type etc. The mutation data contains the molecular information about genes for each cancer patient. It describes various mutations found in the genes of the cancer patients. This includes name of the genes mutated, mutation classification, mutation type etc.   
Finally, we downloaded the gene expression data which contains the level of gene expression estimated from RNA samples of each patient. TCGA has tumor samples for each patient and where possible, non-tumor/normal samples are also present. Thus, we have gene expression data for both tumor and normal tissues, although the number of normal samples are much less. Instead of directly using gene expression data from TCGA, we downloaded data from Wang et. al. 2018 [@wang2018unifying]. This data was created by processing the TCGA data. This data was normalized and corrected for batch effects and thus is comparable accross different samples.


### Downloading clinical data from TCGA

We used the TCGABiolinks [@colaprico2015tcgabiolinks] package to download the TCGA clinical data for BRCA studies. The function *GDCquery_clinic* was used to download the data. We wrote additional functions to clean the data and arrange it into a dataframe.

```{r,message=FALSE,warning=FALSE,results="hide"}
#author urmi
library(TCGAbiolinks)
library(DT)
library(data.table)
library(plyr)
library(maftools) 
library(gganatogram)
library(viridis)
library(gridExtra)
library(grid)
library(RColorBrewer)



colsToKeep<-c("clinical.submitter_id","clinical.classification_of_tumor",
              "clinical.primary_diagnosis","clinical.tumor_stage",
              "clinical.age_at_diagnosis","clinical.vital_status",
              "clinical.days_to_death","clinical.tissue_or_organ_of_origin",
              "clinical.days_to_birth","clinical.site_of_resection_or_biopsy",
              "clinical.days_to_last_follow_up","clinical.cigarettes_per_day",
              "clinical.weight","clinical.alcohol_history","clinical.bmi",
              "clinical.years_smoked","clinical.height","clinical.gender",
              "clinical.year_of_birth","clinical.race","clinical.ethnicity",
              "clinical.year_of_death","clinical.bcr_patient_barcode",
              "clinical.disease","submitter_id","sample_type","tissue_type",
              "portions.submitter_id","portions.analytes.analyte_type",
              "portions.analytes.submitter_id","portions.analytes.analyte_type_id",
              "portions.analytes.aliquots.analyte_type","portions.analytes.aliquots.submitter_id")
#Function takes a df and expands it by unlisting elements at a column
expand<-function(df,colName){
  res<-data.frame()
  #for each row
  for(i in 1: dim(df)[1]){
    thisRow<-df[i, ! (colnames(df) %in% c(colName))]
    tempdf<-as.data.frame(df[i, c(colName)])
    #if list is empty skip that row
    if(dim(tempdf)[1]<1){
      next
    }
    #change colnames so they are unique
    colnames(tempdf)<-paste(paste(colName,".",sep = ""),colnames(tempdf),sep = "")
    #print(paste(i,colnames(tempdf)))
    newRow<-cbind(thisRow,tempdf,row.names = NULL)
    res<-bind_rows(res,newRow)
    
  }
  #print(res)
  return(res)
}

getjoinedBiospcClinc<-function(projName){
  print(paste("Downloading",projName))
  clinicalBRCA <- GDCquery_clinic(project = projName, type = "clinical")
  biospecimenBRCA <- GDCquery_clinic(project = projName, type = "Biospecimen")
  
  #rename all cols from clinical table with suffix clinical
  colnames(clinicalBRCA)<- paste0("clinical.",colnames(clinicalBRCA))
  
  #expand biospecimen data in the order portions, portions.analytes, portions.analytes.aliquots
  toUnpack<-c("portions", "portions.analytes", "portions.analytes.aliquots")
  for(s in toUnpack){
    biospecimenBRCA<-expand(biospecimenBRCA,s)
  }
  #add patient barcode to biospecimen data
  biospecimenBRCA<- biospecimenBRCA %>% mutate(clinical.bcr_patient_barcode=substr(submitter_id,1,nchar(as.character(submitter_id))-4))
  #join clinical and biospecimen
  finalJoined<-join(clinicalBRCA,biospecimenBRCA,by="clinical.bcr_patient_barcode")
  return(finalJoined)
}
###################################################################
projName<-"BRCA"
##Download only BRCA metadata
brcaDF<-getjoinedBiospcClinc(paste("TCGA",projName,sep = "-"))
brcaDF<-brcaDF[,colsToKeep]
#remove cols with all NA values
naCols<-colnames(brcaDF)[sapply(brcaDF, function(x)all(is.na(x)))]
brcaDF<-brcaDF[,!(colnames(brcaDF) %in% naCols)]

```


### Downloading mutation data from TCGA

Next, we downloaded mutation data from BRCA studies using TCGABiolinks' *GDCquery_Maf* function [@colaprico2015tcgabiolinks]. After getting the mutation data, we joined the clinical table with the mutation table to have clinical information for each mutation in the mutation table.

```{r,message=FALSE,warning=FALSE,results="hide"}

brcaMAF <- GDCquery_Maf(projName, pipelines = "varscan2")
#join mutation information with the clinical data
colnames(brcaDF)[which(colnames(brcaDF)==
                         "portions.analytes.aliquots.submitter_id")]="Tumor_Sample_Barcode"
brcaMAF_MD<-join(brcaMAF,brcaDF)
```

This resulted in a data frame of dimensions `r dim(brcaMAF_MD)`. This data frame contains clinical information and coressponding mutation information for each BRCA patient. 

### Downloading gene expression data

Next, we downloaded the gene expression data for the BRCA studies from [@wang2018unifying]. 

```{r,message=FALSE,warning=FALSE,results="hide"}
#author urmi
#download exp data from github for BRCA samples. reading files can take several minutes
brcaexp_nontumor <- read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/brca-rsem-fpkm-tcga.txt.gz",
                               "\t", escape_double = FALSE, trim_ws =TRUE)
brcaexp_tumor <- read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/brca-rsem-fpkm-tcga-t.txt.gz", 
                            "\t", escape_double = FALSE, trim_ws = TRUE)

#formatting. add row names remove extra columns
rn<-brcaexp_nontumor$Hugo_Symbol
brcaexp_nontumor<-brcaexp_nontumor[,3:dim(brcaexp_nontumor)[2]]
rownames(brcaexp_nontumor)<-rn

rn<-brcaexp_tumor$Hugo_Symbol
brcaexp_tumor<-brcaexp_tumor[,3:dim(brcaexp_tumor)[2]]
rownames(brcaexp_tumor)<-rn

```

These datasets contains the gene expression patterns over the tumor samples and normal samples. Each row corresponds to a gene and each column corresponds to a sample. The data dimentions for tumor samples was `r dim(brcaexp_tumor)` and for normal samples was `r dim(brcaexp_nontumor)`.  


# Analysis

We did some exploratory data analysis on our data in order to find associations between various clinical attributes, gene mutations, and gene expression.

## Highly mutated genes

First, we looked at the mutation data to find which genes are highly mutated in BRCA and what types of different mutations are present in the data. We used the package *maftools*[@mayakonda2018maftools] to plot a summary of the BRCA mutation dataset.

```{r,message=FALSE,warning=FALSE,results="hide",echo=FALSE}
#author urmi
#function to take a list of maf files and summarize them
plotmafSummaryList <- function(mafList) {
l1 <- mafList
#for each item in list do calculations
lnames <- names(l1)
plotList <- list()
k <- 1
plist <- list()
for (i in lnames) {
print((i))
print(dim(l1[[i]]))
if (dim(l1[[i]])[1] < 1) {
next
}
#plot summary and save to pdf
maf <- read.maf(l1[[i]], isTCGA = T)
plotmafSummary(
maf = maf,
rmOutlier = TRUE,
addStat = 'median',
dashboard = T,
titvRaw = FALSE,
showBarcodes = F,
top = 10
)
mtext(
paste("Mutation summary by", i),
outer = T,
cex = NA,
line = -1.0,
side = 3
)
}

}
#split maf file into categories by a given variable in the clinical metadata file
splitMafby <- function(clinicalData, by, mafData) {
#get tumor samps fo different values of by
myList <- list()
uniqVals <- clinicalData %>% select(by) %>% unique
for (i in 1:nrow(uniqVals)) {
#i<-1
s <- as.character(uniqVals[i, 1])
print(s)
tList <-
clinicalData %>% filter(clinicalData[, by] == s) %>% select(Tumor_Sample_Barcode) %>%
unique
thisData <-
mafData %>% filter(Tumor_Sample_Barcode %in% tList$Tumor_Sample_Barcode)
myList[[s]] <- thisData
}

return(myList)
}

#function to plot top mutated genes with ggplot
plotGeneVarFreq <- function(mafList) {
  #for each item in list
  lnames <- names(mafList)
  plotList <- list()
  k <- 1
  plist <- list()
  getPalette = colorRampPalette(brewer.pal(9, "Paired"))
  # assign colors manually to be consistent with missing data
  variants <- c(
    "Missense_Mutation",
    "Silent",
    "3'UTR",
    "Nonsense_Mutation",
    "5'Flank",
    "Intron",
    "Splice_Region",
    "RNA",
    "5'UTR",
    "Splice_Site",
    "In_Frame_Del",
    "Frame_Shift_Ins",
    "Frame_Shift_Del",
    "In_Frame_Ins",
    "3'Flank",
    "Nonstop_Mutation",
    "Translation_Start_Site",
    "IGR"
  )
  colourCount = length(unique(variants))
  palette <- getPalette(colourCount)
  names(palette) <- variants
  
  for (i in lnames) {
    print((i))
    print(dim(mafList[[i]]))
    if (dim(mafList[[i]])[1] < 1) {
      next
    }
    
    #find top mutated genes
    thisMaf <- mafList[[i]]
    topGenes <-
      thisMaf %>% filter(Variant_Classification != "Silent") %>% select(Hugo_Symbol) %>% group_by(Hugo_Symbol) %>% count %>% arrange(desc(freq)) %>% top_n(n=10)
    maftest <- thisMaf %>% filter(Hugo_Symbol %in% topGenes$Hugo_Symbol & Variant_Classification != "Silent") %>% select(Hugo_Symbol, Variant_Classification)
    
    
    p <-
      ggplot(data = maftest, aes(x = Hugo_Symbol, fill = Variant_Classification)) +
      geom_bar(stat = "count") + scale_x_discrete(limits = rev(topGenes$Hugo_Symbol)) +
      coord_flip() +
      theme(legend.position = "right") +
      theme(
        axis.text.x = element_text(size = 11, face = "bold"),
        axis.text.y = element_text(size = 11, face = "bold"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(),
        axis.title = element_text(size = 12, face = "bold")
      ) +
      ylab("") + xlab("") + scale_fill_manual(values = palette) + ggtitle(i)
    
    plist[[k]] <- p
    k = k + 1
    
  }
  #arrange plots on a grid
  do.call("grid_arrange_shared_legend", c(plist))
  
}

#plot on a grid with single legend
#function reference 
#https://stackoverflow.com/questions/13649473/add-a-common-legend-for-combined-ggplots
grid_arrange_shared_legend <- function(...) {
plots <- list(...)
g <-
ggplotGrob(plots[[1]] + theme(legend.position = "bottom"))$grobs
legend <- g[[which(sapply(g, function(x)
x$name) == "guide-box")]]
lheight <- sum(legend$height)
grid.arrange(
do.call(arrangeGrob, lapply(plots, function(x)
x + theme(legend.position = "none"))),
legend,
ncol = 1,
heights = unit.c(unit(1, "npc") - lheight, lheight)
)
}

```

```{r fig1, fig.width=8, fig.height=8,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig1}BRCA mutations summary"}
#author urmi
#plot summary for full dataset
tempList<-list()
tempList[["BRCA"]]<-brcaMAF
plotmafSummaryList(tempList)

```

Figure \ref{fig:fig1} shows the summary plot generated by maftools. From the plot we can see that majority of mutations are of type missense mutation. Missense mutations can change the amino acid sequence of the protein coded by the mutated gene making an unstable protein product. After missense mutations, we see a lot of nonsense mutations. A gene with nonsense mutation produces a truncated protein which is non-functional. We also found that, most variant types are Single Neucleotide Polymorphisms (SNPs). These variants are a result of mutation of only a single neucleotide in the DNA. SNPs can have deletarious effects if they result in missense or nonsense mutations. We also found the top 10 genes with highest number of mutations. Out of these genes PIK3CA, TP53, CDH1 and PTEN are already known to be associated with cancer.  TP53, for example, codes for a tumor suppressor protein which is essential for regulation of cell cycle. Any mutation in this gene can cause changes in the TP53 protein which may not function as a tumor suppressor anymore thus causing tumor.

Next, we looked at how the gene mutations differ for different categories. For this task we wrote our own function to plot summary as barchart. First, we looked at which genes were mutated in samples from different race. 

```{r fig2, fig.width=8, fig.height=10,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig2}Top mutated genes by race"}
#author urmi
#split the maf data by clinical.race and plot summary for each category
l1<-splitMafby(brcaDF,"clinical.race",brcaMAF)
#plot only white black and asian as others are not reported
l1<-l1[c("white","black or african american","asian" )]
plotGeneVarFreq(l1)

```

Figure \ref{fig:fig2} shows the top mutated genes for each race.  PIK3CA and TP53 genes are the most mutated for each category. There are unique genes too for each category e.g. MUC16 for white, DNM1p47 for black/african american and C5orf42 for asian populations. We also found majority of mutations are of type missense mutation for each category.   

Then, we looked at how genes are mutated for the two most common breast cancer types i.e. ductal carcinoma and lobular carcinoma.

```{r fig3, fig.width=8, fig.height=8,cache=FALSE,warning=FALSE,message=FALSE,results="hide",fig.cap="\\label{fig:fig3}Top mutated genes by cancer type"}
#author urmi
#split the maf data by clinical.race and plot summary for each category
filterList<- brcaDF %>% select(clinical.primary_diagnosis) %>% count %>% arrange(desc(freq)) %>% top_n(n=2)
l2<-splitMafby(brcaDF%>%filter(clinical.primary_diagnosis %in% filterList$clinical.primary_diagnosis),"clinical.primary_diagnosis",brcaMAF)
plotGeneVarFreq(l2)
```

Figure \ref{fig:fig3} shows the top mutated genes for ductal carcinoma and lobular carcinoma. Interestingly, the gene TP53 has lesser mutations in lobular carcinoma that in ductal carcinoma.

From the analysis above we noticed that BRCA1 and BRCA2 are not in the set of top mutated genes. These genes are known to be associated with cancer and a mutation in these genes can increase the risk of breast and ovarian cancer in women. To find out how many mutations belonged to BRCA1 and BRCA2 in our dataset, we plotted a barchart (Figure \ref{fig:fig3}).

```{r fig4, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig4}BRCA1/BRCA2 mutation statistics"}
#author urmi
brcaMAF_MD_geneGroupBRCA<-brcaMAF_MD%>%select(Hugo_Symbol)%>%mutate(geneName=ifelse(Hugo_Symbol=="BRCA1","BRCA1",ifelse(Hugo_Symbol=="BRCA2","BRCA2","Others"))) %>%select(geneName)%>% group_by(geneName) %>% count %>% mutate(logFreq=log(freq))

p1<-ggplot(data=brcaMAF_MD_geneGroupBRCA, aes(x=geneName,y=logFreq,fill=geneName))+geom_bar(stat = "identity")+theme(legend.position = "none")+
  theme(axis.text.x = element_text(size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_brewer(palette="Paired")



p2<-ggplot(data=brcaMAF_MD%>%filter(Hugo_Symbol %in% c("BRCA1","BRCA2")),aes(x=Hugo_Symbol,fill=Variant_Classification))+geom_bar()+
  theme(axis.text.x = element_text(size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),legend.text=element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_brewer(palette="Dark2")

grid.arrange(p1, p2, nrow = 1)

```

Figure \ref{fig:fig4} shows the number of mutations (log scale) in BRCA1/2 and all other genes combined and the type of mutation for BRCA1 and BRCA2 genes. We found that mutations in these two genes were very small as compared to other genes. Missense mutation is most common in these two genes too. 

## Expression patterns of genes

After looking at various mutated genes, we then looked at the expression pattern of these genes. We looked at the distributions of gene expression for tumor and normal tissues. First, we took the set of highly mutated genes in BRCA and compared their expression in tumor and normal samples.

```{r fig5, fig.width=8, fig.height=6,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig5}Expression of highly mutated genes in tumor and normal samples"}
#author urmi
#plot box plots of top mutated genes
topGenes<-c("PIK3CA","TP53","TTN","GATA3","CDH1","MAP3K1","MUC16","KMT2C","MUC4","PTEN")
nt<-as.data.frame(t(brcaexp_nontumor[topGenes,]))
colnames(nt)<-topGenes
#tumordata
tdata<-as.data.frame(t(brcaexp_tumor[topGenes,]))
colnames(tdata)<-topGenes
nt<-nt%>%mutate(source="Normal")
tdata<-tdata%>%mutate(source="Tumor")

topCombined<-bind_rows(nt,tdata)
ggplot(melt(topCombined), aes(x=factor(variable),y=value,fill=factor(source)))+geom_boxplot(outlier.colour=NA)+scale_y_log10()+theme(legend.position = "top",legend.text = element_text(size = 15,face = "bold"))+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ 
  scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")


```

Figure \ref{fig:fig5} shows the boxplots for the ten genes. We can see visually that expression of genes TP53, CDH1, and MAP3K1 looks very simillar and unchanged in the two categories. While the expression of genes PIK3CA, TTN, GATA3 and MUC4 looks different under the two conditions. This might give some insight on how these genes play a role in cancer. A statistical test is still required to assess the significance of these differences.   

After that, we compiled a list of genes are known to be differentially expressed in cancer [@li2017transcriptional]. Genes which are differntially expressed could be important biomarkers and give us more insights about the most affected biological pathways in cancer. 

```{r fig6, fig.width=10, fig.height=5,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig6}Expression of genes known to be regulated in tumor and normal samples"}
#author urmi
#ref for data https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5470989/
upGenes<-c("E2F1", "EZH2", "FOXM1", "MYBL2", "PLK1")
dwnGenes<-c("SCARA5", "MYOM1", "NKAPL", "PEG3", "USP2")


ntUP<-as.data.frame(t(brcaexp_nontumor[upGenes,]))
colnames(ntUP)<-upGenes
ntUP<-ntUP%>%mutate(source="Non-Tumor")
tdataUP<-as.data.frame(t(brcaexp_tumor[upGenes,]))
colnames(tdataUP)<-upGenes
tdataUP<-tdataUP%>%mutate(source="Tumor")

diffExpgenes<-bind_rows(ntUP,tdataUP)
dodge <- position_dodge(width = 0.5)

p1<-ggplot(melt(diffExpgenes), aes(x=factor(variable),y=value,fill=factor(source)))+geom_violin(position = dodge)+geom_boxplot(width=.1, outlier.colour=NA, position = dodge)+scale_y_log10()+theme(legend.position = "top")+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")

#for down regulated in cancer
ntDWN<-as.data.frame(t(brcaexp_nontumor[dwnGenes,]))
colnames(ntDWN)<-dwnGenes
ntDWN<-ntDWN%>%mutate(source="Non-Tumor")
tdataDWN<-as.data.frame(t(brcaexp_tumor[dwnGenes,]))
colnames(tdataDWN)<-dwnGenes
tdataDWN<-tdataDWN%>%mutate(source="Tumor")
diffExpgenesDWN<-bind_rows(ntDWN,tdataDWN)


p2<-ggplot(melt(diffExpgenesDWN), aes(x=factor(variable),y=value,fill=factor(source)))+geom_violin(position = dodge)+geom_boxplot(width=.1, outlier.colour=NA, position = dodge)+scale_y_log10()+theme(legend.position = "top")+
  theme(axis.text.x = element_text(angle=45,size = 15,face = "bold"),axis.text.y = element_text(size = 10,face = "bold"),panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        panel.border = element_blank(),
        panel.background = element_blank(), axis.title=element_text(size=12,face="bold"))+ scale_fill_discrete(name = "Gene")+ylab("log(expression)")+xlab("")+scale_fill_brewer(palette="Set1")

grid.arrange(p1, p2, nrow = 1)

```

Figure \ref{fig:fig6} shows violin plots for the cancer regulated genes. From the plots we can see that all the genes on the left panel appear to be downregulated and all the genes in the right panel seems to be up regulated in BRCA tumor samples as compared to normal samples. This result was expected and also mentioned in @li2017transcriptional.

## Comparing expression of mutated BRCA genes with other cancer types

Finally, we wanted to compare the expression of highly mutated genes in BRCA with other cancers from different tissues. To do this we downloaded more expression data for four tumors from different tissues which were colon, liver, lung, and stomach. Then we looked at the mean expression of each of these genes accross different tissues. To visualize the mean expression over different tissues we used the package gganatogram [@maag2018gganatogram].


```{r,message=FALSE,warning=FALSE,results="hide"}
#author urmi
#download more expression data
#read tumor data for brca,liver,stomach,colon,Lung
brca <- brcaexp_tumor

coad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/coad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
coadRownames<-coad$Hugo_Symbol
coad<-coad[,3:dim(coad)[2]]
rownames(coad)<-coadRownames

lihc<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/lihc-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
lihcRownames<-lihc$Hugo_Symbol
lihc<-lihc[,3:dim(lihc)[2]]
rownames(lihc)<-lihcRownames

luad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/luad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
luadRownames<-luad$Hugo_Symbol
luad<-luad[,3:dim(luad)[2]]
rownames(luad)<-luadRownames

stad<-read_delim("https://raw.githubusercontent.com/mskcc/RNAseqDB/master/data/normalized/stad-rsem-fpkm-tcga.txt.gz", "\t", escape_double = FALSE, trim_ws = TRUE)
stadRownames<-stad$Hugo_Symbol
stad<-stad[,3:dim(stad)[2]]
rownames(stad)<-stadRownames

topGenes<-c("PIK3CA","TP53","GATA3","CDH1","MAP3K1","PTEN")
expVals<-c()
temp<-t(brca[topGenes,])
colnames(temp)<-topGenes
meanbrca<-colMeans(temp)

temp<-t(coad[topGenes,])
colnames(temp)<-topGenes
meancoad<-colMeans(temp)

temp<-t(lihc[topGenes,])
colnames(temp)<-topGenes
meanlihc<-colMeans(temp)

temp<-t(luad[topGenes,])
colnames(temp)<-topGenes
meanluad<-colMeans(temp)

temp<-t(stad[topGenes,])
colnames(temp)<-topGenes
meanstad<-colMeans(temp)
```

```{r fig7, fig.width=10, fig.height=8,cache=FALSE,warning=FALSE,message=FALSE,results="hide", fig.cap="\\label{fig:fig7}Mean expression of highly mutated BRCA genes in tumors from different tissues"}
#author urmi
organs<-c("breast","colon","liver","lung","stomach")
type<-c( "other","digestion","digestion","respiratory","digestion")
colour<-c("#41ab5d","orange","orange","steelblue","orange")


plots<-list() 
for(i in 1:6){
  vals<-c(meanbrca[i],meancoad[i],meanlihc[i],meanluad[i],meanstad[i])
  gganatogramData<-data.frame(organ=organs,type=type,colour=colour,value=as.numeric(vals),stringsAsFactors=F)
  
  p<-gganatogram(data=gganatogramData, fillOutline='#a6bddb', organism='human', sex='female', fill="value")+ 
    theme_void()+  scale_fill_gradient(low = "yellow", high = "red",name= paste(topGenes[i],"(fpkm)"))+theme(legend.text = element_text(size=15,face = "bold"),legend.position = c(0.75, 0.2))
  
  plots[[i]]<-p
}

n <- length(plots)
nCol <- floor(sqrt(n))
do.call("grid.arrange", c(plots, ncol=3))

```

Figure \ref{fig:fig7} shows the mean expression of the genes in tumors from different tissues. We found that BRCA samples had higher mean expression as compared with other tissues.  A definite conclusion requires a statistical test though.


# Conclusion

From looking at the cancer statistics we can say that cancer is a growing problem. 
The statistics may healp researchers, doctors and policy makers

In order to find more effective methods to treat cancer, we must first fully understand the disease itself. A number of mutations in one's DNA can lead to cancer. We saw that in BRCA few genes get mutated most often like PIK3CA and TP53. We already know that mutations in these genes results in an altered protein which is not capable of normal biological function. We also need to better understand how these mutation affects the cellular pathways. Expression patterns of these genes may reveal if they are getting activated or suppressed due to cancer. Genes which are differentially regulated in cancer may prove to be important biomarkers. Studying functions of these genes may lead to new ways of stopping cancer in the body.  


# References


